util/metrics/tagcounter.go

```diff
@@ -98,32 +102,34 @@ func (tc *TagCounter) WriteMetric(buf *strings.Builder, parentLabels string) {
 		// no values, nothing to say.
 		return
 	}
-	// TODO: what to do with "parentLabels"? obsolete part of interface?
-	buf.WriteString("# ")
-	buf.WriteString(tc.Name)
-	buf.WriteString(" ")
-	buf.WriteString(tc.Description)
-	buf.WriteString("\n")
 	isTemplate := strings.Contains(tc.Name, "{TAG}")
 	tags := tagptr.(map[string]*uint64)
 	for tag, tagcount := range tags {
 		if tagcount == nil {
 			continue
 		}
+		var name string
 		if isTemplate {
-			name := strings.ReplaceAll(tc.Name, "{TAG}", tag)
-			buf.WriteString(name)
-			buf.WriteRune(' ')
-			buf.WriteString(strconv.FormatUint(*tagcount, 10))
-			buf.WriteRune('\n')
+			name = strings.ReplaceAll(tc.Name, "{TAG}", tag)
 		} else {
-			buf.WriteString(tc.Name)
-			buf.WriteRune('_')
-			buf.WriteString(tag)
-			buf.WriteRune(' ')
-			buf.WriteString(strconv.FormatUint(*tagcount, 10))
-			buf.WriteRune('\n')
+			name = tc.Name + "_" + tag
+		}
+		buf.WriteString("# HELP ")
```

Comment:
What is the value of adding HELP here?

Reply:
the HELP line provides a description of the metric for some UIs & backends use ... for example Grafana 6.6+ shows it alongside the metric name when you are browsing metrics https://grafana.com/blog/2020/06/15/how-we-made-working-with-prometheus-easier-with-metric-metadata-in-grafanas-explore-view/